nixos/modules/services/home-automation/homebridge.nix

```diff
@@ -0,0 +1,549 @@
+            type = submodule {
+              freeformType = settingsFormat.type;
+              options = {
+                method = mkOption {
+                  type = str;
+                  default = "systemd";
+                  description = "Method to use for logging";
+                  readOnly = true;
+                };
+
+                service = mkOption {
+                  type = str;
+                  default = "homebridge";
+                  description = "Name of the systemd service to log to";
+                  readOnly = true;
+                };
+              };
+            };
+          };
+
+          # The following options are allowed to be changed.
+          port = mkOption {
+            type = port;
+            default = 8581;
+            description = "The port the UI web service should listen on";
+          };
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.homebridge = {
+      description = "Homebridge";
+      wants = [ "network-online.target" ];
+      after = [
+        "syslog.target"
+        "network-online.target"
+      ];
+      wantedBy = [ "multi-user.target" ];
+
+      # On start, if the config file is missing, create a default one
+      # Otherwise, ensure that the config file is using the
+      # properties as specified by nix.
+      # Not sure if there is a better way to do this than to use jq
+      # to replace sections of json.
+      preStart = ''
+        # If the user storage path does not exist, create it
+        if [ ! -d "${cfg.userStoragePath}" ]; then
+          install -d -m 700 -o ${cfg.user} -g ${cfg.group} "${cfg.userStoragePath}"
+        fi
+        # If there is no config file, create a placeholder default
+        if [ ! -e "${cfg.userStoragePath}/config.json" ]; then
+          install -D -m 600 -o ${cfg.user} -g ${cfg.group} "${defaultConfigFile}" "${cfg.userStoragePath}/config.json"
+        fi
+
+        # Write the jq merge filter file to a temporary file
+        install -D -m 600 -o ${cfg.user} -g ${cfg.group} "${jqMergeFilterFile}" "${cfg.userStoragePath}/jqMergeFilter.jq"
+
+        # Write the nix override config to a temporary file
+        install -D -m 600 -o ${cfg.user} -g ${cfg.group} "${nixOverrideConfigFile}" "${cfg.userStoragePath}/nixOverrideConfig.json"
+
+        # Apply all nix override settings to config.json in a single jq operation
+        ${pkgs.jq}/bin/jq -s -f "${cfg.userStoragePath}/jqMergeFilter.jq" "${cfg.userStoragePath}/config.json" "${cfg.userStoragePath}/nixOverrideConfig.json" | ${pkgs.jq}/bin/jq . > "${cfg.userStoragePath}/config.json.tmp"
+        install -D -m 600 -o ${cfg.user} -g ${cfg.group} "${cfg.userStoragePath}/config.json.tmp" "${cfg.userStoragePath}/config.json"
+
+        # Remove temporary files
+        rm "${cfg.userStoragePath}/jqMergeFilter.jq"
+        rm "${cfg.userStoragePath}/nixOverrideConfig.json"
+        rm "${cfg.userStoragePath}/config.json.tmp"
```

Comment:
I don't think you need to copy these to temporary files. You should be able to use e.g. `jqMergeFilterFile` directly

Reply:
Consolidated these, to reference the files directly

Reply:
Great!